core/src/test/resources/configuration/smart-testing.yml

```diff
@@ -13,3 +13,13 @@ autocorrect: true #<!--9-->
 customStrategies:  #<!--10-->
     - smart.testing.strategy.cool=org.arquillian.smart.testing:strategy-cool:1.0.0
     - smart.testing.strategy.experimental=org.arquillian.smart.testing:strategy-experimental:1.0.0
+strategiesConfiguration: #<!--11-->
+    affected:
+      config: affected-configuration.properties #<!--12-->
```

Comment:
I think that the way of using affected related config file should be replaced by this new implementation - in other words, there should be only one config file for everything.

Reply:
no currently it's not implemented this way. We are reading only `exclusions` & `inclusions` from config file. We are not reading `transitivity` in config file. may be something to improve later.

Reply:
what do you mean by the fact that the `transitivity` is not read in a config file? Which config file? I can see that everything is loaded in this new implementation (including `transitivity` parameter): https://github.com/arquillian/smart-testing/pull/243/files#diff-b222e33537ad5f62b9ddccd10f1d89d9R19

The purpose of this PR is to have one configuration file for both the general configuration and all strategy related configurations. There shouldn't be any additional configuration file for the strategies...

Reply:
you can define eveything in configuration file using current approach. But along with this you can define `inclusions` & `exclusions` using properties file.

Also if we remove this `properties` file support & ask user to give all `inclusions` & `exclusions` in yaml file, then user has to define all includes & excludes in config file, then while overwriting inclusions & exclusions using system property then user has to write whole list of inclusions & exclusions. That's why it's good to provide support for both.

Reply:
> you can define eveything in configuration file using current approach. But along with this you can define inclusions & exclusions using properties file.

I would really welcome one consistent way of defining the properties. This means a usage of one config file or the system properties. Not several configuration files. We had this discussion before and IIRC we agreed on having it in one config file. Am I right @bartoszmajsak @lordofthejars ?

> Also if we remove this properties file support & ask user to give all inclusions & exclusions in yaml file, then user has to define all includes & excludes in config file, 

Yes, this is what we want - to have everything in one place. It is not expected that the includes and excludes will be somehow long...

> then while overwriting inclusions & exclusions using system property then user has to write whole list of inclusions & exclusions.

But this will be same also for another external property file - you need to either change the properties file or define everything using system properties...
BTW. introducing another properties file brings another complication - what takes the precedence? Values specified
* in a properties file
* in YAML config file
* in a properties file that is set by a system property
* by a system property. 

I would vote for keeping it consistent and simple - having one config file.

Reply:
Ideally we'll be taking everything specified in properties file I think we are not overwriting anything in properties file.

then whatever we are defining in yaml configuration file we are combining it with properties file.

However we are overwriting `inclusions` & `exclusion` defined in yaml configuration file using system properties.